test/format/snap.c

```diff
@@ -114,9 +116,11 @@ snap_op_init(TINFO *tinfo, uint64_t read_ts, bool repeatable_reads)
         /*
          * If the stable timestamp has changed and we've advanced beyond it, preserve the current
          * snapshot history up to this point, we'll use it verify rollback_to_stable. Switch our
-         * tracking to the other snap list.
+         * tracking to the other snap list. Use a barrier to ensure a cached value doesn't cause us
+         * to ignore a stable timestamp transition.
          */
-        stable_ts = __wt_atomic_addv64(&g.stable_timestamp, 0);
+        WT_BARRIER();
+        stable_ts = g.stable_timestamp;
```

Comment:
I prefer `WT_ORDERED_READ` instead of a `WT_BARRIER` here, I think the semantic is more understandable.

Reply:
I don't believe this is a case of `WT_ORDERED_READ` -- it's been awhile since I looked at this, but my recollection is the concern that a locally cached value could be wrong, not that the read ordering can create a problem. Anyway, I'd be cautious with this change.